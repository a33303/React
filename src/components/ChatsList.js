import React, {useState} from 'react'
import {nanoid} from "nanoid"
import {ListItemButton, ListItemText} from "@mui/material";

export function ChatsList() {
    const [chats, setChats] = useState([
        {id: nanoid(), name: "По интересам"},
        {id: nanoid(), name: "Общий"},
        {id: nanoid(), name: "О Всяком"},
    ])

    return (
        chats.map((i) =>
            <ListItemButton key={i.id}>
                <ListItemText primary={i.name} />
            </ListItemButton>)
    )

}



