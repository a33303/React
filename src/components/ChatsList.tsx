import React, { useState } from "react";
import { nanoid } from "nanoid";
import { ListItemButton, ListItemText } from "@mui/material";

export const ChatsList = () => {
    const [chats, setChats] = useState([
        {id: nanoid(), name: "Interesting"},
        {id: nanoid(), name: "Everything"},
        {id: nanoid(), name: "Countries"},
    ])

    return (
        chats.map((i) =>
            <ListItemButton key={i.id}>
                <ListItemText primary={i.name} />
            </ListItemButton>)
    )
}



