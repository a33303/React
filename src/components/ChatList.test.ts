import { ChatsList } from "./ChatsList"
import renderer from 'react-test-renderer'
import React from "react"
import { render, screen } from "@testing-library/react"

describe('createChatList', () => {
    it('to be Function ', function () {
        expect(ChatsList).toBeInstanceOf(Function)
    })
    it('renders from component - ChatsList ', function () {
        const tree = renderer
            .create(<ChatsList />)
        expect(tree).toMatchSnapshot()

    })
})