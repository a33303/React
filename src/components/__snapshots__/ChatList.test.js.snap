// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createChatList renders from component - ChatsList  1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-1kqdo7n-MuiListItemButton-root{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;-webkit-box-pack:start;-ms-flex-pack:start;-webkit-justify-content:flex-start;justify-content:flex-start;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;position:relative;-webkit-text-decoration:none;text-decoration:none;box-sizing:border-box;text-align:left;padding-top:8px;padding-bottom:8px;-webkit-transition:background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;padding-left:16px;padding-right:16px;}.css-1kqdo7n-MuiListItemButton-root:hover{-webkit-text-decoration:none;text-decoration:none;background-color:rgba(0, 0, 0, 0.04);}@media (hover: none){.css-1kqdo7n-MuiListItemButton-root:hover{background-color:transparent;}}.css-1kqdo7n-MuiListItemButton-root.Mui-selected{background-color:rgba(25, 118, 210, 0.08);}.css-1kqdo7n-MuiListItemButton-root.Mui-selected.Mui-focusVisible{background-color:rgba(25, 118, 210, 0.2);}.css-1kqdo7n-MuiListItemButton-root.Mui-selected:hover{background-color:rgba(25, 118, 210, 0.12);}@media (hover: none){.css-1kqdo7n-MuiListItemButton-root.Mui-selected:hover{background-color:rgba(25, 118, 210, 0.08);}}.css-1kqdo7n-MuiListItemButton-root.Mui-focusVisible{background-color:rgba(0, 0, 0, 0.12);}.css-1kqdo7n-MuiListItemButton-root.Mui-disabled{opacity:0.38;}",
      }
    }
    data-emotion="css 1kqdo7n-MuiListItemButton-root"
  />,
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;position:relative;box-sizing:border-box;-webkit-tap-highlight-color:transparent;background-color:transparent;outline:0;border:0;margin:0;border-radius:0;padding:0;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;vertical-align:middle;-moz-appearance:none;-webkit-appearance:none;-webkit-text-decoration:none;text-decoration:none;color:inherit;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;-webkit-box-pack:start;-ms-flex-pack:start;-webkit-justify-content:flex-start;justify-content:flex-start;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;position:relative;-webkit-text-decoration:none;text-decoration:none;box-sizing:border-box;text-align:left;padding-top:8px;padding-bottom:8px;-webkit-transition:background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;padding-left:16px;padding-right:16px;}.css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root::-moz-focus-inner{border-style:none;}.css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root.Mui-disabled{pointer-events:none;cursor:default;}@media print{.css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root{-webkit-print-color-adjust:exact;color-adjust:exact;}}.css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root:hover{-webkit-text-decoration:none;text-decoration:none;background-color:rgba(0, 0, 0, 0.04);}@media (hover: none){.css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root:hover{background-color:transparent;}}.css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root.Mui-selected{background-color:rgba(25, 118, 210, 0.08);}.css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root.Mui-selected.Mui-focusVisible{background-color:rgba(25, 118, 210, 0.2);}.css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root.Mui-selected:hover{background-color:rgba(25, 118, 210, 0.12);}@media (hover: none){.css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root.Mui-selected:hover{background-color:rgba(25, 118, 210, 0.08);}}.css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root.Mui-focusVisible{background-color:rgba(0, 0, 0, 0.12);}.css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root.Mui-disabled{opacity:0.38;}",
      }
    }
    data-emotion="css cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
  />,
  <div
    className="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-tlelie-MuiListItemText-root{-webkit-flex:1 1 auto;-ms-flex:1 1 auto;flex:1 1 auto;min-width:0;margin-top:4px;margin-bottom:4px;}",
        }
      }
      data-emotion="css tlelie-MuiListItemText-root"
    />
    <div
      className="MuiListItemText-root css-tlelie-MuiListItemText-root"
    >
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-10hburv-MuiTypography-root{margin:0;font-family:\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif;font-weight:400;font-size:1rem;line-height:1.5;letter-spacing:0.00938em;display:block;}",
          }
        }
        data-emotion="css 10hburv-MuiTypography-root"
      />
      <span
        className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
      >
        По интересам
      </span>
    </div>
  </div>,
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-1kqdo7n-MuiListItemButton-root{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;-webkit-box-pack:start;-ms-flex-pack:start;-webkit-justify-content:flex-start;justify-content:flex-start;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;position:relative;-webkit-text-decoration:none;text-decoration:none;box-sizing:border-box;text-align:left;padding-top:8px;padding-bottom:8px;-webkit-transition:background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;padding-left:16px;padding-right:16px;}.css-1kqdo7n-MuiListItemButton-root:hover{-webkit-text-decoration:none;text-decoration:none;background-color:rgba(0, 0, 0, 0.04);}@media (hover: none){.css-1kqdo7n-MuiListItemButton-root:hover{background-color:transparent;}}.css-1kqdo7n-MuiListItemButton-root.Mui-selected{background-color:rgba(25, 118, 210, 0.08);}.css-1kqdo7n-MuiListItemButton-root.Mui-selected.Mui-focusVisible{background-color:rgba(25, 118, 210, 0.2);}.css-1kqdo7n-MuiListItemButton-root.Mui-selected:hover{background-color:rgba(25, 118, 210, 0.12);}@media (hover: none){.css-1kqdo7n-MuiListItemButton-root.Mui-selected:hover{background-color:rgba(25, 118, 210, 0.08);}}.css-1kqdo7n-MuiListItemButton-root.Mui-focusVisible{background-color:rgba(0, 0, 0, 0.12);}.css-1kqdo7n-MuiListItemButton-root.Mui-disabled{opacity:0.38;}",
      }
    }
    data-emotion="css 1kqdo7n-MuiListItemButton-root"
  />,
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;position:relative;box-sizing:border-box;-webkit-tap-highlight-color:transparent;background-color:transparent;outline:0;border:0;margin:0;border-radius:0;padding:0;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;vertical-align:middle;-moz-appearance:none;-webkit-appearance:none;-webkit-text-decoration:none;text-decoration:none;color:inherit;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;-webkit-box-pack:start;-ms-flex-pack:start;-webkit-justify-content:flex-start;justify-content:flex-start;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;position:relative;-webkit-text-decoration:none;text-decoration:none;box-sizing:border-box;text-align:left;padding-top:8px;padding-bottom:8px;-webkit-transition:background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;padding-left:16px;padding-right:16px;}.css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root::-moz-focus-inner{border-style:none;}.css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root.Mui-disabled{pointer-events:none;cursor:default;}@media print{.css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root{-webkit-print-color-adjust:exact;color-adjust:exact;}}.css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root:hover{-webkit-text-decoration:none;text-decoration:none;background-color:rgba(0, 0, 0, 0.04);}@media (hover: none){.css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root:hover{background-color:transparent;}}.css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root.Mui-selected{background-color:rgba(25, 118, 210, 0.08);}.css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root.Mui-selected.Mui-focusVisible{background-color:rgba(25, 118, 210, 0.2);}.css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root.Mui-selected:hover{background-color:rgba(25, 118, 210, 0.12);}@media (hover: none){.css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root.Mui-selected:hover{background-color:rgba(25, 118, 210, 0.08);}}.css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root.Mui-focusVisible{background-color:rgba(0, 0, 0, 0.12);}.css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root.Mui-disabled{opacity:0.38;}",
      }
    }
    data-emotion="css cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
  />,
  <div
    className="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-tlelie-MuiListItemText-root{-webkit-flex:1 1 auto;-ms-flex:1 1 auto;flex:1 1 auto;min-width:0;margin-top:4px;margin-bottom:4px;}",
        }
      }
      data-emotion="css tlelie-MuiListItemText-root"
    />
    <div
      className="MuiListItemText-root css-tlelie-MuiListItemText-root"
    >
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-10hburv-MuiTypography-root{margin:0;font-family:\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif;font-weight:400;font-size:1rem;line-height:1.5;letter-spacing:0.00938em;display:block;}",
          }
        }
        data-emotion="css 10hburv-MuiTypography-root"
      />
      <span
        className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
      >
        Общий
      </span>
    </div>
  </div>,
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-1kqdo7n-MuiListItemButton-root{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;-webkit-box-pack:start;-ms-flex-pack:start;-webkit-justify-content:flex-start;justify-content:flex-start;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;position:relative;-webkit-text-decoration:none;text-decoration:none;box-sizing:border-box;text-align:left;padding-top:8px;padding-bottom:8px;-webkit-transition:background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;padding-left:16px;padding-right:16px;}.css-1kqdo7n-MuiListItemButton-root:hover{-webkit-text-decoration:none;text-decoration:none;background-color:rgba(0, 0, 0, 0.04);}@media (hover: none){.css-1kqdo7n-MuiListItemButton-root:hover{background-color:transparent;}}.css-1kqdo7n-MuiListItemButton-root.Mui-selected{background-color:rgba(25, 118, 210, 0.08);}.css-1kqdo7n-MuiListItemButton-root.Mui-selected.Mui-focusVisible{background-color:rgba(25, 118, 210, 0.2);}.css-1kqdo7n-MuiListItemButton-root.Mui-selected:hover{background-color:rgba(25, 118, 210, 0.12);}@media (hover: none){.css-1kqdo7n-MuiListItemButton-root.Mui-selected:hover{background-color:rgba(25, 118, 210, 0.08);}}.css-1kqdo7n-MuiListItemButton-root.Mui-focusVisible{background-color:rgba(0, 0, 0, 0.12);}.css-1kqdo7n-MuiListItemButton-root.Mui-disabled{opacity:0.38;}",
      }
    }
    data-emotion="css 1kqdo7n-MuiListItemButton-root"
  />,
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;position:relative;box-sizing:border-box;-webkit-tap-highlight-color:transparent;background-color:transparent;outline:0;border:0;margin:0;border-radius:0;padding:0;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;vertical-align:middle;-moz-appearance:none;-webkit-appearance:none;-webkit-text-decoration:none;text-decoration:none;color:inherit;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;-webkit-box-pack:start;-ms-flex-pack:start;-webkit-justify-content:flex-start;justify-content:flex-start;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;position:relative;-webkit-text-decoration:none;text-decoration:none;box-sizing:border-box;text-align:left;padding-top:8px;padding-bottom:8px;-webkit-transition:background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;padding-left:16px;padding-right:16px;}.css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root::-moz-focus-inner{border-style:none;}.css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root.Mui-disabled{pointer-events:none;cursor:default;}@media print{.css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root{-webkit-print-color-adjust:exact;color-adjust:exact;}}.css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root:hover{-webkit-text-decoration:none;text-decoration:none;background-color:rgba(0, 0, 0, 0.04);}@media (hover: none){.css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root:hover{background-color:transparent;}}.css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root.Mui-selected{background-color:rgba(25, 118, 210, 0.08);}.css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root.Mui-selected.Mui-focusVisible{background-color:rgba(25, 118, 210, 0.2);}.css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root.Mui-selected:hover{background-color:rgba(25, 118, 210, 0.12);}@media (hover: none){.css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root.Mui-selected:hover{background-color:rgba(25, 118, 210, 0.08);}}.css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root.Mui-focusVisible{background-color:rgba(0, 0, 0, 0.12);}.css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root.Mui-disabled{opacity:0.38;}",
      }
    }
    data-emotion="css cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
  />,
  <div
    className="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-tlelie-MuiListItemText-root{-webkit-flex:1 1 auto;-ms-flex:1 1 auto;flex:1 1 auto;min-width:0;margin-top:4px;margin-bottom:4px;}",
        }
      }
      data-emotion="css tlelie-MuiListItemText-root"
    />
    <div
      className="MuiListItemText-root css-tlelie-MuiListItemText-root"
    >
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-10hburv-MuiTypography-root{margin:0;font-family:\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif;font-weight:400;font-size:1rem;line-height:1.5;letter-spacing:0.00938em;display:block;}",
          }
        }
        data-emotion="css 10hburv-MuiTypography-root"
      />
      <span
        className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
      >
        О Всяком
      </span>
    </div>
  </div>,
]
`;
